modules/ROOT/pages/patterns/shortest-paths.adoc

```diff
@@ -538,3 +538,84 @@ RETURN [stop IN nodes(p) | stop.name] AS r1,
 
 2+d|Rows: 1
 |===
+
+
+[[planning-StatefulShortestPath-operator]]
+== Planning the StatefulShortestPath-operator
+See xref:planning-and-tuning/operators/operators-detail.adoc#query-plan-stateful-shortest-path-into[the operator-detail page] for a description of the `StatefulShortestPath(All)` and `StatefulShortestPath(Into)` operators.
+
+When the subexpression of the statefulShortestPath-operator matches only one of both boundary nodes, then a `StatefulShortestPath(All)` is planned.
+
+When the subexpression of the statefulShortestPath-operator matches both boundary nodes, then a `StatefulShortestPath(Into)` is planned only when both boundary nodes match to at most one data node. Otherwise, this plan is avoided when possible.
+Another plan will be favoured where the subexpression matches only one of both boundary nodes followed by an `StatefulShortestPath(All)`.
```

Comment:
I find these paragraphs confusing. 

```
When the subexpression of the statefulShortestPath-operator matches only one of both boundary nodes
```

When would that happen?
What is "the subexpression of the statefulShortestPath-operator"?

---

I would phrase it along these lines:

```
The planner decides between the two variants using this heuristic:

* When both boundary nodes having been matched in a `MATCH` clause prior to the `MATCH` clause containing the shortest paths query, `StatefulShortestPath(Into)` is planned.
* Otherwise, if both boundary nodes match to at most one data node, `StatefulShortestPath(Into)` is planned.
* Otherwise, `StatefulShortestPath(All)` is planned.

Sometimes the planner cannot make reliable estimations about how many data nodes a patern nodes will match. Consider using a uniqueness constraint <insert link there> where applicable to help the planner getting more reliable estimates.
```